django_mesh/models.py

```diff
@@ -191,4 +194,48 @@ class MockYoutube(models.Model):
     you = models.CharField(max_length=140, unique=True)
     yoh = models.CharField(max_length=140, unique=True)
     yot = models.TextField(default='')
-    yoj = models.TextField(default='')
\ No newline at end of file
+    yoj = models.TextField(default='')
+
+class Media(_Abstract):
+
+    MEDIA_TYPE = Choices(
+        (0, 'NONE',       'none',),
+        (1, 'IMAGE',      'image',),
+        (2, 'VIDEO',      'video',),
+    )
+
+    channel = models.ForeignKey(Channel)
+    objects = MediaQuerySet.as_manager()
+
+
+    upload_file = models.FileField(upload_to='uploads/%Y/%m/%d')
+
+    media_type = models.IntegerField(max_length=1, default=MEDIA_TYPE.NONE, choices=MEDIA_TYPE)
+
+
+
+    oembed_html = models.TextField(default='', blank=True)
+
+    @property    
+    def get_file_url(self):
+        return self.upload_file.url
+
+    def render(self):
+
+        embed = '<a href="'+self.get_file_url+'"> %s </a>' % self.get_file_url
+
+        if self.media_type == Media.MEDIA_TYPE.IMAGE:
+            embed = '<img src="'+self.get_file_url+'" style="width:100%;height:100%;>'
+
+        elif self.media_type == Media.MEDIA_TYPE.VIDEO:
+            embed = '<video width="700" height="350" controls> <source src="'+self.get_file_url+'" type="video/mp4"> Your browser does not support the video tag. </video>'
+
+        self.oembed_html = embed
+
+    def get_absolute_url(self):
+        return reverse('mesh_media_view', kwargs={'slug': self.slug,})
+
+    def save(self, *args, **kwargs):
```

Comment:
I think this shouldn't be needed since we call render() in the parent class' save().